src/templates/ReportForm.tsx

```diff
@@ -77,6 +77,48 @@ const ReportForm = ({ data, location }) => {
   const [hasContestEnded, setHasContestEnded] = useState<boolean>(
     Date.now() > new Date(end_time).getTime()
   );
+  const [ipState, setIpState] = useState<{
+    warden: string;
+    hash: string;
+  }>();
+  const [currentIp, setCurrentIp] = useState<string>("");
+
+  const checkIP = useCallback(async () => {
+    try {
+      const res = await fetch("https://api.ipify.org?format=json");
+      const data = await res.json();
+      const currentIP = data.ip;
+      setCurrentIp(currentIP);
+      const ipObj = window.localStorage.getItem("hash");
+      const ip = CryptoJS.AES.encrypt(
+        currentIP,
+        process.env.GATSBY_CRYPTO_ENCRYPTION_KEY!
```

Comment:
iiuc this `GATSBY_CRYPTO_ENCRYPTION_KEY` is public, since everyone who loads the site would have it clientside?